homeassistant/components/xiaomi_miio/air_quality.py

```diff
@@ -41,52 +47,54 @@
 
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
-    """Set up the sensor from config."""
+    """Import Miio configuration from YAML."""
+    _LOGGER.warning(
+        "Loading Xiaomi Miio Air Quality via platform setup is deprecated. "
+        "Please remove it from your configuration."
+    )
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data=config,
+        )
+    )
 
-    host = config[CONF_HOST]
-    token = config[CONF_TOKEN]
-    name = config[CONF_NAME]
 
-    _LOGGER.info("Initializing with host %s (token %s...)", host, token[:5])
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Xiaomi Air Quality from a config entry."""
+    entities = []
 
-    miio_device = Device(host, token)
+    if config_entry.data[CONF_FLOW_TYPE] == CONF_DEVICE:
+        host = config_entry.data[CONF_HOST]
+        token = config_entry.data[CONF_TOKEN]
+        name = config_entry.title
+        model = config_entry.data[CONF_MODEL]
+        unique_id = config_entry.unique_id
 
-    try:
-        device_info = await hass.async_add_executor_job(miio_device.info)
-    except DeviceException as ex:
-        raise PlatformNotReady from ex
+        _LOGGER.debug("Initializing with host %s (token %s...)", host, token[:5])
 
-    model = device_info.model
-    unique_id = f"{model}-{device_info.mac_address}"
-    _LOGGER.debug(
-        "%s %s %s detected",
-        model,
-        device_info.firmware_version,
-        device_info.hardware_version,
-    )
+        device = AirQualityMonitor(host, token, model=model)
 
-    device = AirQualityMonitor(host, token, model=model)
+        if model == MODEL_AIRQUALITYMONITOR_S1:
+            entities.append(AirMonitorS1(name, device, config_entry, unique_id))
+        elif model == MODEL_AIRQUALITYMONITOR_B1:
+            entities.append(AirMonitorB1(name, device, config_entry, unique_id))
+        elif model == MODEL_AIRQUALITYMONITOR_V1:
+            entities.append(AirMonitorV1(name, device, config_entry, unique_id))
+        else:
+            raise NoEntitySpecifiedError(f"Not support for entity {unique_id}")
```

Comment:
We can remove this case, I think.

Reply:
I changed it to a _LOGGER.warning, just in case some new model comes allong in the future it will be esier for people to identify why the entity is not beeing set up.